code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -107,8 +116,10 @@ public boolean doWork(final Event event) {
      */
     protected OptimizeExtension(final ExtensionApi extensionApi) {
         super(extensionApi);
-
-        cachedPropositions = new HashMap<>();
+        /// Concurrent Map containing the get propositions (and corresponding requested scopes) for
+        // Edge events that are yet to be pushed to the Edge network.
+        // This is accessed from multiple threads.
+        cachedPropositions = new ConcurrentHashMap<>();
```

Comment:
To maintain consistency, could you move this initialization and comment to the declaration on Line 39?

Reply:
done please check @praveek